src/test/java/com/example/onlinebookstore/service/CategoryServiceTest.java

```diff
@@ -0,0 +1,135 @@
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@ExtendWith(MockitoExtension.class)
+public class CategoryServiceTest {
+
+    @Mock
+    private CategoryRepository categoryRepository;
+
+    @InjectMocks
+    private CategoryServiceImpl categoryService;
+
+    @Spy
+    private CategoryMapper categoryMapper = new CategoryMapperImpl();
+
+    @Test
+    @DisplayName("Verify the correct category returns when category exists")
+    public void getCategoryById_WithValidId_ShouldReturnValidCategory() {
+        Long categoryId = 1L;
+
+        Category category = new Category();
+        category.setId(categoryId);
+        category.setName("Adults");
+
+        Mockito.when(categoryRepository.findById(categoryId)).thenReturn(Optional.of(category));
+
+        CategoryResponseDto actual = categoryService.getById(categoryId);
+
+        CategoryResponseDto expected = new CategoryResponseDto(1L, "Adults", null);
+
+        Assertions.assertTrue(EqualsBuilder.reflectionEquals(expected, actual));
+    }
+
+    @Test
+    @DisplayName("Verify the correct exception message returns with not valid id")
+    public void getCategoryById_WithNotValidId_ShouldReturnException() {
+        Long categoryId = 1000L;
+
+        Mockito.when(categoryRepository.findById(categoryId)).thenReturn(Optional.empty());
+
+        Exception exception = Assertions.assertThrows(EntityNotFoundException.class,
+                () -> categoryService.getById(categoryId));
+
+        String expected = "Can't find category by this categoryId: " + categoryId;
+
+        String actual = exception.getMessage();
+
+        Assertions.assertEquals(expected, actual);
+    }
+
+    @Test
+    @DisplayName("Verify the correct bookDto returns after book was saved")
+    public void saveCategory_Valid_ShouldReturnCategoryResponseDto() {
+        Long categoryId = 1L;
+
+        Category category = new Category();
+        category.setId(categoryId);
+        category.setName("Adults");
+        category.setDescription("For adults");
+
+        CategoryRequestDto requestDto = new CategoryRequestDto("Adults", "For adults");
+
+        Mockito.when(categoryRepository.save(Mockito.any(Category.class))).thenReturn(category);
+
+        CategoryResponseDto actual = categoryService.save(requestDto);
+        CategoryResponseDto expected = new CategoryResponseDto(categoryId, "Adults", "For adults");
+
+        Assertions.assertTrue(EqualsBuilder.reflectionEquals(expected, actual));
+    }
+
+    @Test
+    @DisplayName("Verify correct delete method invocation")
+    public void deleteCategoryById_ValidId_Success() {
+        Long id = 1L;
+
+        Mockito.doNothing().when(categoryRepository).deleteById(id);
+
+        categoryService.deleteById(id);
+
+        Mockito.verify(categoryRepository, times(1)).deleteById(id);
```

Comment:
```suggestion
        Mockito.verify(categoryRepository).deleteById(id);
```
times(1) is a default value, so you can skip it